File: template/go/template_method_test.go
package template

import (
	"fmt"
	"testing"
)

func TestTemplateMethod(t *testing.T) {
	recipient := NewRecipient("张三", "13800138000", "zhangsan@email.com", "地球村")
	message := "这是一条有用的消息，请查阅！"

	s := NewSMSSender()
	s.Send(recipient, message)

	fmt.Println("==========================================")

	s = NewEmailSender()
	s.Send(recipient, message)

	fmt.Println("==========================================")

	s = NewLetterSender()
	s.Send(recipient, message)
}

File: template/java/TemplateMethodTest.java
public class TemplateMethodTest {
    public static void main(String[] args) {
        Recipient recipient = new Recipient("张三", "13800138000", "zhangsan@email.com", "地球村");
        String message = "这是一条有用的消息，请查阅！";

        ISender sender = new SMSSender();
        sender.send(recipient, message);

        System.out.println("==========================================");

        sender = new EmailSender();
        sender.send(recipient, message);

        System.out.println("==========================================");

        sender = new LetterSender();
        sender.send(recipient, message);
    }
}